common/src/db/query.rs

```diff
@@ -420,18 +611,34 @@ mod tests {
     #[test(tokio::test)]
     async fn sorts() -> Result<(), anyhow::Error> {
         // Good sorts
-        assert!(Sort::<advisory::Entity>::from_str("location").is_ok());
-        assert!(Sort::<advisory::Entity>::from_str("location:asc").is_ok());
-        assert!(Sort::<advisory::Entity>::from_str("location:desc").is_ok());
-        assert!(Sort::<advisory::Entity>::from_str("Location").is_ok());
-        assert!(Sort::<advisory::Entity>::from_str("Location:Asc").is_ok());
-        assert!(Sort::<advisory::Entity>::from_str("Location:Desc").is_ok());
+        assert!(Sort::from_str_with_entity::<advisory::Entity>("location").is_ok());
+        assert!(Sort::from_str_with_entity::<advisory::Entity>("location:asc").is_ok());
+        assert!(Sort::from_str_with_entity::<advisory::Entity>("location:desc").is_ok());
+        assert!(Sort::from_str_with_entity::<advisory::Entity>("Location").is_ok());
+        assert!(Sort::from_str_with_entity::<advisory::Entity>("Location:Asc").is_ok());
+        assert!(Sort::from_str_with_entity::<advisory::Entity>("Location:Desc").is_ok());
```

Comment:
To me, the new name is awkward, perhaps because it's replacing a `FromStr` impl? I can't decide if introducing some new type, e.g. `SortedEntity` with its own `from_str` is an improvement or not.